keras/src/trainers/trainer.py

```diff
@@ -325,8 +328,12 @@ def metrics(self):
             loss = self._compile_loss(y, y_pred, sample_weight)
             if loss is not None:
                 losses.append(loss)
-        for loss in self.losses:
-            losses.append(ops.sum(ops.cast(loss, dtype=backend.floatx())))
+
+        # If in symbolic scope, skip `self.losses` to ensure we don't access
+        # any variables. Otherwise, it might break.
+        if not in_symbolic_scope():
```

Comment:
This feels quite brittle -- `compute_loss` is a method that is intended to be overridden by users in various cases, and those users should not need to be aware of `SymbolicScope`. Why is it necessary?

Reply:
I have moved the logic to `Layer._get_regularization_losses`.
This should be more robust now.